pkg/ipfix/ipfix_intermediate_test.go

```diff
@@ -0,0 +1,93 @@
+package ipfix
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/go-ipfix/pkg/entities"
+	ipfixintermediate "github.com/vmware/go-ipfix/pkg/intermediate"
+)
+
+var (
+	fields = []string{
+		"sourcePodName",
+		"sourcePodNamespace",
+		"sourceNodeName",
+		"destinationPodName",
+		"destinationPodNamespace",
+		"destinationNodeName",
+		"destinationClusterIPv4",
+		"destinationClusterIPv6",
+		"destinationServicePort",
+		"ingressNetworkPolicyRuleAction",
+		"egressNetworkPolicyRuleAction",
+		"ingressNetworkPolicyRulePriority",
+	}
+)
+
+const (
+	testActiveExpiry   = 100 * time.Millisecond
+	testInactiveExpiry = 150 * time.Millisecond
+	testMaxRetries     = 2
+)
+
+func TestNewIPFIXAggregationProcess(t *testing.T) {
+	tc := []struct {
+		input            ipfixintermediate.AggregationInput
+		expectedNumFlows int64
+		expectedError    string
+		testname         string
+	}{
+		{
+			testname: "Message channel not initialized",
+			input: ipfixintermediate.AggregationInput{
+				MessageChan:     nil,
+				WorkerNum:       2,
+				CorrelateFields: fields,
+			},
+			expectedError: "error while initializing IPFIX intermediate process: cannot create AggregationProcess process without message channel",
+		},
+		{
+			testname: "Message channel initialized",
+			input: ipfixintermediate.AggregationInput{
+				MessageChan:     make(chan *entities.Message),
+				WorkerNum:       2,
+				CorrelateFields: fields,
+			},
+			expectedNumFlows: 0,
+			expectedError:    "",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.testname, func(t *testing.T) {
+			aggregationProcess, err := NewIPFIXAggregationProcess(tt.input)
+			if tt.expectedError == "" {
+				assert.Equal(t, tt.expectedNumFlows, aggregationProcess.GetNumFlows())
+			} else {
+				assert.Equal(t, tt.expectedError, err.Error())
+			}
+		})
+	}
+}
+
```

Comment:
few functions like SetCorrelatedFieldsFilled in ipfix_intermediate.go , yet to be covered !